src/msw/renderer.cpp

```diff
@@ -1107,20 +1107,31 @@ void wxRendererXP::DrawGauge(wxWindow* win,
     ::GetThemeBackgroundContentRect(
         hTheme,
         GetHdcOf(dc.GetTempHDC()),
-        PP_BAR,
+        flags & wxCONTROL_SPECIAL ? PP_BARVERT : PP_BAR,
         0,
         &r,
         &contentRect);
 
-    contentRect.right = contentRect.left +
+    if ( flags & wxCONTROL_SPECIAL )
+    {
+        // For a vertical gauge, the value grows from the bottom to the top.
+        contentRect.top = contentRect.bottom -
+                          wxMulDivInt32(contentRect.bottom - contentRect.top,
+                                        value,
+                                        max);
+    }
+    else // Horizontal.
+    {
+        contentRect.right = contentRect.left +
                             wxMulDivInt32(contentRect.right - contentRect.left,
                                           value,
                                           max);
+    }
 
     ::DrawThemeBackground(
         hTheme,
         GetHdcOf(dc.GetTempHDC()),
-        PP_CHUNK,
+        flags & wxCONTROL_SPECIAL ? PP_FILLVERT : PP_FILL,
```

Comment:
Should be `PP_CHUNK` and `PP_CHUNKVERT`?

Reply:
I do notice some:
![pp_chunk](https://user-images.githubusercontent.com/8088070/35770015-43919606-0914-11e8-94c8-57f525840a49.png)
![pp_fill](https://user-images.githubusercontent.com/8088070/35770016-45bb0534-0914-11e8-802f-9928ea7c39ff.png)

An extra pixel between the border and the filled area. I guess the former (chunk) is XP style, the latter (fill) Vista+.

Reply:
Hmm, this is not what I see here at all, why doesn't it use the gradient in your screenshots?

Reply:
This is what my normal Windows 10 progress bar looks like as well. It doesn't have a gradient*.

*only a gradient moving from left to right every few seconds so it looks like there is progress.

Reply:
Sorry, I didn't realize it changed since Windows 7 (which I still primarily use). We should probably use whichever looks best under Windows 10, so should we switch to `PP_FILL`, finally?

Lack of animation is a problem, of course as is lack of "hot" state support in e.g. `DrawTreeItemButton()`), but this is a different story...

Reply:
I compared it to the progress dialog when copying, and the progress dialog in CMake, both are entirely filled (like `PP_CHUNK`). I don't think we need to switch.

Reply:
Thanks for testing! There could still be a reason to switch to `PP_FILL`, see [this idea](http://trac.wxwidgets.org/ticket/18076), but I'm definitely not going to do it right now.